WordPress/src/main/java/org/wordpress/android/ui/activitylog/list/filter/ActivityLogTypeFilterViewModel.kt

```diff
@@ -1,20 +1,123 @@
 ) : ScopedViewModel(mainDispatcher) {
     private var isStarted = false
 
+    private val _uiState = MutableLiveData<UiState>()
+    val uiState: LiveData<UiState> = _uiState
+
     fun start() {
         if (isStarted) return
         isStarted = true
+
+        _uiState.value = FullscreenLoading
+        fetchAvailableActivityTypes()
+    }
+
+    private fun fetchAvailableActivityTypes() {
+        launch {
+            // TODO malinjir initiate the fetch
+            onActivityTypesFetched(immutableListOf(DummyActivityType, DummyActivityType, DummyActivityType))
+        }
+    }
+
+    private suspend fun onActivityTypesFetched(activityTypes: List<DummyActivityType>) {
+        _uiState.value = buildContentUiState(activityTypes)
+    }
+
+    private suspend fun buildContentUiState(activityTypes: List<DummyActivityType>): Content {
+        return withContext(bgDispatcher) {
+            // TODO malinjir replace the hardcoded header title
+            val headerListItem = SectionHeader(UiStringText("Test"))
+            // TODO malinjir replace "it.toString()" with activity type name
+            val activityTypeListItems: List<ListItemUiState.ActivityType> = activityTypes
+                    .map {
+                        ListItemUiState.ActivityType(title = UiStringText(it.toString()))
+                    }
+            Content(
+                    listOf(headerListItem) + activityTypeListItems,
+                    primaryAction = Action(label = UiStringRes(R.string.activity_log_activity_type_filter_apply))
+                            .apply { action = ::onApplyClicked },
+                    secondaryAction = Action(label = UiStringRes(R.string.activity_log_activity_type_filter_clear))
+                            .apply { action = ::onClearClicked }
+            )
+        }
+    }
+
+    private fun onApplyClicked() {
+        // TODO malinjir save and dismiss
+    }
+
+    private fun onClearClicked() {
+        (_uiState.value as? Content)?.let { it ->
+            _uiState.value = it.copy(items = uncheckAllActivityTypeItems(it))
+        }
+    }
+
+    private fun uncheckAllActivityTypeItems(it: Content): List<ListItemUiState> {
+        return it.items.map { item ->
+            if (item is ListItemUiState.ActivityType) {
+                item.copy(checked = false)
+            } else {
+                item
+            }
+        }
+    }
+
+    sealed class UiState {
+        open val contentVisibility = false
+        open val loadingVisibility = false
+
+        object FullscreenLoading : UiState() {
+            override val loadingVisibility: Boolean = true
+        }
+
+        data class Content(
+            val items: List<ListItemUiState>,
+            val primaryAction: Action,
+            val secondaryAction: Action
+        ) : UiState() {
+            override val contentVisibility = true
+        }
+
+        data class Action(val label: UiString) {
```

Comment:
I decided to keep this as is. The only reason the parameter is not in the constructor is that two lambdas are not equal unless they reference the exact same instance. I want to use equals in the RV diffUtil, however, implementing a custom equals felt like an overkill.

Reply:
Totally, I think we should not move it to constructor. The main purpose of this change is about renaming it to something else to avoid confusion. The changing it to a constructor is a secondary though and might not be applicable usually.